src/components/common/dashboard/OperationComponent.vue

```diff
@@ -0,0 +1,386 @@
+                        <button
+                            :id="'op_reject_' + shownOpId"
+                            :title="writtenReason == '' ? 'Please provide a reason' : 'Reject'"
+                            :disabled="writtenReason == ''"
+                            class="btn btn-icon-red-filled text-sm h-12"
+                            @click.stop="reject"
+                        >
+                            Reject
+                        </button>
+                        <button
+                            :id="'op_cancelRejection_' + shownOpId"
+                            class="ml-2 btn btn-icon-blue text-sm h-12"
+                            @click.stop="toogleReasonMenu"
+                        >
+                            Cancel
+                        </button>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+    import Operation from "@/api/api_models/operation_management/Operation";
+    import { computed, onBeforeMount, ref, watch } from "vue";
+    import { OperationStatus } from "@/api/api_models/operation_management/OperationState";
+    import { useStore } from "@/use/store/store";
+    import { MutationTypes } from "@/use/store/mutation-types";
+    import { RejectionReasons } from "./reasons";
+    import { UC4Identifier } from "@/api/helpers/UC4Identifier";
+    import { showNotYetImplementedToast } from "@/use/helpers/Toasts";
+    import { Role } from "@/entities/Role";
+    import CertificateManagement from "@/api/CertificateManagement";
+    import GenericResponseHandler from "@/use/helpers/GenericResponseHandler";
+    import Router from "@/use/router";
+    import { approveMatriculation, rejectOperation } from "@/api/abstractions/FrontendSigning";
+    import { printOperation, printOperationTitle } from "@/api/abstractions/OperationPrinter";
+
+    export default {
+        name: "OperationComponent",
+        components: {},
+        props: {
+            operation: {
+                type: Object as () => Operation,
+                required: true,
+            },
+            enrollmentId: {
+                type: String,
+                required: true,
+            },
+            role: {
+                type: String,
+                required: true,
+            },
+            isArchive: {
+                type: Boolean,
+                required: false,
+                default: false,
+            },
+            watched: {
+                type: Boolean,
+                required: false,
+            },
+        },
+        emits: ["marked-read"],
+        setup(props: any, { emit }: any) {
+            const store = useStore();
+            const operation = ref(props.operation as Operation);
+            const approvals = operation.value.existingApprovals.users.length + operation.value.existingApprovals.groups.length;
+            const neededApprovals =
```

Comment:
required

Reply:
change in https://github.com/upb-uc4/ui-web/pull/785/commits/1796e5816def599553396b1941fe815cf60fae0b